### DISCUSSION TITLE
[suggestion] gravity auto-bounds

### DISCUSSION THREAD

**Discussion by amplified1** [tsunamayo/Starship-EVO#3941] [Suggestions]
basically it would be a button that when pressed changes the settings for a gravity generator so that it's bounds are perfectly squared around the ship, the user could then make manual adjustments to make it perfect but it would make setting up gravity slightly less tedious

**Comment by tsunamayo** [comment]
@amplified1 Hi, I am not sure to understand... gravity has a maximum size. Maybe do you want it to not extend beyond the ship bound?

**Comment by ZachZent** [comment]
At this point we just need a shipwide gravity net generator to the ship's bounds. Neovox allowing ships to get to a stupidly large size, be much simpler for the player to place, power, and leave it to work as intended. 16x16x16 individual ranges in 500m+ ships and stations and you will fly out of a ship without them, it is a massive pain no matter how user friendly the controls are.

**Comment by amplified1** [comment]
yeah I think it makes more sense to just have a really large max value and just scale the power and heat draw.

**Comment by cloudrambler** [comment]
Agree we need a larger / full ship gravity generator, but still don't like the idea of it just being a rectangle, as that would leave gravity outside ship in some areas. I don't want gravity throwing me off my ship if I try to access the exterior hull.

**Comment by CurioInventorium** [comment]
This could be a coding nightmare to do, but how about gravity decals? They could work along side the existing gravity block system.

Since the character is about 2m tall, each decal would only project a three meter high gravity area above it (or along the normal of the decal). So that the character jumping wouldn't immediately jump out of the field.

Stretch the decal to 3m x 4m on the floor, and you get a 3 x 4 x 3 (high) gravity area.
The default 3m high would also help with placing blocks on top of the decals.

Place the decal on the wall and you've got sideways gravity.

Whether they can be hooked up to the logic system or not, or some sort of global grid power the same way turrets are powered is a coding issue I suppose.

**Comment by amplified1** [comment]
I think that would possibly cause issues as the game moves into a survival stage since it doesn't really make sense to have a sticker work like the gravity generator, you also wouldn't be able to change the intensity of the gravity, toggle it, or anything

**Comment by CurioInventorium** [comment]
That's why I mentioned that they would work along side the existing gravity block. These would be useful in small areas, or long catwalks where a standard gravity requirement and no config needed.
The power to toggle them could be tied into the global grid power.

The idea's been used in a Sci-Fi series of books by Neil Asher. Where in some ship/station areas they use gravity floor tiles rather than gravity generators.

In-game there's already a system in place for setting the armour level of a block just by using the paint system, so why not use decals for controlling where gravity is.

I agree with your point on having larger gravity areas. It really drags down my enthusiasm for doing a large build if I have place (and configure) gravity blocks at regular intervals. The range needs to be similar to the shipyard blocks range.

**Comment by tsunamayo** [comment]
I am trying the new discussion tool with that suggestion.
Okay I will work on this soon. I dont link using decals. I could give:
- a new block that does shipwide gravity
- allow the current generator to have a larger field.
- allow the current generator to have a toggle: shipwide gravity.

I need to make up my mind on the pros and cons of each solution. Again I want to see how it fit in a survival context.

Also working with field is painful, I still want to have something that you can drag and rescale in game to be more seamless.

**Reply by Garrett-C** [reply]
That sounds pretty good yeah. With larger ships being much more viable now it does make sense to have something like shipwide gravity. I know I am terrified of trying to add gravity to the venator and ISD lol. 

Would definitely be nice to also have a way of adding smaller field within it that overrides the shipewide one. So that you can have some gravity shenanigans when you want.

**Reply by ProPeach** [reply]
A combination of the shipwide gravity toggle in the existing block as well as an easier way to rescale the field would be perfect, thanks for looking into this! The new Discussions feature is a great idea too

**Comment by Uncle-Ulty** [comment]
We need some improvement for gravity block for sure. Big structures require tons of gravity blocks. With a bigger bound, it would save several blocks. 

The following racing track has 2130 gravity blocks.

![image](https://user-images.githubusercontent.com/73253472/124760759-6f4cc780-df07-11eb-8ff3-8488b5d41642.png)
